### PR TITLE
SCIF proof-of-concept

### PR DESCRIPTION
**Pull request by julianpistorius**
See: https://github.com/AgPipeline/issues-and-projects/issues/132

### CODE REVIEW

File: soilmask.py [hunk 1]
```diff
@@ -34,6 +50,151 @@ def gen_plant_mask(color_img: np.ndarray, kernel_size: int = 3) -> np.ndarray:
...
+            if nodata:
+                output_raster.GetRasterBand(band).SetNoDataValue(nodata)
+    else:
+        # single channel image, e.g. temperature
+        output_raster.GetRasterBand(1).WriteArray(pixels)
+        output_raster.GetRasterBand(1).FlushCache()
+        if nodata:
+            output_raster.GetRasterBand(1).SetNoDataValue(nodata)
+
+    output_raster = None
```
Comment (julianpistorius) [inline soilmask.py:197]: @Chris-Schnaufer This creates `output_raster` and then doesn't do anything with it. Is that right?

File: soilmask.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    blur = cv2.blur(mask, (kernel_size, kernel_size))
+    pix = np.array(blur)
+    sub_mask = pix > 128
+
+    mask_1 = np.zeros_like(b_channel)
+    mask_1[sub_mask] = MAX_PIXEL_VAL
+
+    return mask_1
+
+def get_epsg(filename):
```
Comment (julianpistorius) [inline soilmask.py:53]: Would it be possible to add Python 3 type annotations like the other functions have?

File: soilmask.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        max_x = max(ulx, lrx)
+
+        return [min_y, max_y, min_x, max_x]
+    except Exception as ex:
+        logging.info("[image_get_geobounds] Exception caught: %s", str(ex))
+
+    return [np.nan, np.nan, np.nan, np.nan]
+
+
+def create_geotiff(pixels, gps_bounds, out_path, nodata=-99, asfloat=False, extractor_info=None, system_md=None,
```
Comment (julianpistorius) [inline soilmask.py:111]: Also missing type annotations, and these parameters are not used at all:

- `extractor_info`
- `system_md`
- `extra_metadata`

File: soilmask.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+        return proj.GetAttrValue('AUTHORITY', 1)
+    # pylint: disable=broad-except
+    except Exception as ex:
+        logger.warn("[get_epsg] Exception caught: %s", str(ex))
+    # pylint: enable=broad-except
+
+    return None
+
+
+def get_image_file_geobounds(filename):
```
Comment (julianpistorius) [inline soilmask.py:76]: Missing type annotation.

File: soilmask.py (hunk 1 above)
Comment (Chris-Schnaufer) [reply soilmask.py:197]: The `output_raster` is used for its "side effects" of creating a georeferenced image on disk. Does that answer your question? This is direct from terrautils

Comment (julianpistorius) [reply soilmask.py:197]: Sortof. Does that mean that if you don't set it to `None` that there will be memory leakage?

Comment (Chris-Schnaufer) [reply soilmask.py:197]: Not at all https://python-textbok.readthedocs.io/en/1.0/Variables_and_Scope.html. That variable should go out of scope when the function returns and get cleaned up (at some point)

Comment (julianpistorius) [reply soilmask.py:197]: OK. I know some Python libraries with C (or other native) bindings (e.g. [pandas](https://github.com/pandas-dev/pandas/issues?utf8=%E2%9C%93&q=is%3Aissue+memory+)) can have problems with dangling references.

I'm not familiar with this particular library, but I can't think of any other reason why someone would add a line like that unless they wanted to ensure that a leak doesn't happen.

**Pull request merged**